File: Drivo.WebAPI/Services/PaymentsService.cs
using Drivo.Entities;
using Drivo.Responses;
using Drivo.WebAPI.Repositories;

namespace Drivo.WebAPI.Services;

public class PaymentsService
{
    public PaymentsService(PaymentsRepository paymentsRepository, StudentsService studentsService)
    {
        PaymentsRepository = paymentsRepository;
        StudentsService = studentsService;
    }

    private PaymentsRepository PaymentsRepository { get; }
    private StudentsService StudentsService { get; }

    public async Task<List<PaymentEntity>> GetPaymentsAsync()
    {
        return await PaymentsRepository.GetPaymentsAsync();
    }

    public async Task<PaymentEntity> GetPaymentByIdAsync(int paymentId)
    {
        return await PaymentsRepository.GetPaymentByIdAsync(paymentId);
    }

    public async Task<ActionResponse> AddPaymentAsync(PaymentEntity payment)
    {
        return await PaymentsRepository.AddPaymentAsync(payment);
    }

    public async Task<ActionResponse> UpdatePaymentAsync(PaymentEntity payment)
    {
        return await PaymentsRepository.UpdatePaymentAsync(payment);
    }

    public async Task<ActionResponse> RemovePaymentAsync(int paymentId)
    {
        return await PaymentsRepository.RemovePaymentAsync(await GetPaymentByIdAsync(paymentId));
    }
}

File: Drivo/src/app/services/payments.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { firstValueFrom } from 'rxjs';
import { PaymentEntity } from 'src/entities/PaymentEntity';
import { environment } from 'src/environments/environment';
import { ActionResponse } from 'src/responses/action.response';

@Injectable({
  providedIn: 'root'
})
export class PaymentsService {

  constructor(private httpClient: HttpClient) { }

  public async getPaymentsAsync(): Promise<Array<PaymentEntity>> {
    return await firstValueFrom(this.httpClient.get<Array<PaymentEntity>>(`${environment.apiUri}/Payments`));
  }

  public async addPaymentAsync(payment: PaymentEntity): Promise<ActionResponse> {
    return await firstValueFrom(this.httpClient.post<ActionResponse>(`${environment.apiUri}/Payments`, payment));
  }

  public async updatePaymentAsync(payment: PaymentEntity): Promise<ActionResponse> {
    return await firstValueFrom(this.httpClient.put<ActionResponse>(`${environment.apiUri}/Payments`, payment));
  }

  public async deletePaymentAsync(paymentId: number): Promise<ActionResponse> {
    return await firstValueFrom(this.httpClient.delete<ActionResponse>(`${environment.apiUri}/Payments/${paymentId}`));
  }
  
}
